src/core.js

```diff
@@ -689,15 +687,14 @@ define([
         return $modal;
     };
 
-    fn._setup_fancybox = function () {
-        $.fancybox.init();
-        this.$f.find("a.inline").fancybox({
-            hideOnOverlayClick: false,
-            hideOnContentClick: false,
-            enableEscapeButton: false,
-            showCloseButton : true,
-            onClosed: function() {}
-        });
+    fn._showPageSpinner = function() {
+        var spinner = $("<div><div><div></div></div></div>");
+        spinner.addClass("fd-form-saving");
+        $('body').append(spinner);
+    };
+
+    fn._hidePageSpinner = function() {
```

Comment:
Any particular reason you made these "private" (leading underscore)? I generally avoid using that pattern for making things private in Vellum (never quite understood the need for this kind of [fake privacy](https://github.com/dimagi/Vellum/blob/b7903bf/src/base.js#L34-L44)). Instead, if it really needs to be private, define them in the closure scope (don't assign to `fn`).

Incidentally, this does seem like something a plugin might want to override, maybe? Although an argument could be made for keeping it private until we know we need that.